protocol/x/subaccounts/keeper/transfer.go

```diff
@@ -225,16 +236,21 @@ func (k Keeper) TransferFeesToFeeCollectorModule(
 		return err
 	}
 
+	collateralPoolAddr, err := k.GetCollateralPoolFromPerpetualId(ctx, perpetualId)
+	if err != nil {
+		return err
+	}
+
 	// Send coins from `subaccounts` to the `auth` module fee collector account.
-	fromModule := types.ModuleName
-	toModule := authtypes.FeeCollectorName
+	fromModule := collateralPoolAddr
```

Comment:
nit: Maybe use `fromModuleAddr` and `toModuleAddr` now that we're generating addresses?
non-blocking.